ecosystem/gstreamer_plugin/gst_mtl_st20p_rx.c

```diff
@@ -461,25 +498,40 @@ static GstFlowReturn gst_mtl_st20p_rx_create(GstBaseSrc* basesrc, guint64 offset
 
   if (!frame) {
     GST_INFO("Failed to get frame EOS");
-    GST_OBJECT_UNLOCK(src);
     return GST_FLOW_EOS;
   }
 
-  gst_buffer_map(buf, &dest_info, GST_MAP_WRITE);
+  gst_buffer_map(*buffer, &dest_info, GST_MAP_WRITE);
 
-  fill_size = gst_buffer_fill(buf, 0, frame->addr[0], src->frame_size);
-  GST_BUFFER_PTS(buf) = frame->timestamp;
+  gsize fill_size = gst_buffer_fill(*buffer, 0, frame->addr[0], src->frame_size);
+  GST_BUFFER_PTS(*buffer) = frame->timestamp;
 
-  gst_buffer_unmap(buf, &dest_info);
+  gst_buffer_unmap(*buffer, &dest_info);
 
   if (fill_size != src->frame_size) {
     GST_ERROR("Failed to fill buffer");
-    ret = GST_FLOW_ERROR;
-  } else {
-    ret = GST_FLOW_OK;
+    return GST_FLOW_ERROR;
   }
 
   st20p_rx_put_frame(src->rx_handle, frame);
+
+  return GST_FLOW_OK;
+}
+
+static GstFlowReturn gst_mtl_st20p_rx_chain(GstBaseSrc* basesrc, guint64 offset,
+                                            guint length, GstBuffer** buffer) {
+  Gst_Mtl_St20p_Rx* src = GST_MTL_ST20P_RX(basesrc);
+
+  gint ret = GST_FLOW_OK;
```

Comment:
```suggestion
  Gst_Mtl_St20p_Rx* src = GST_MTL_ST20P_RX(basesrc);
  gint ret = GST_FLOW_OK;
```

Reply:
Fixed in bedd0895579ff54984a2df2d2aed7191e446539a